minari/dataset/minari_storage.py

```diff
@@ -328,31 +336,22 @@ def observation_space(self) -> gym.Space:
         """Observation Space of the dataset."""
         if self._observation_space is None:
             with h5py.File(self._file_path, "r") as file:
-                if "observation_space" in file.attrs.keys():
-                    serialized_space = file.attrs["observation_space"]
-                    assert isinstance(serialized_space, str)
-                    self._observation_space = deserialize_space(serialized_space)
-                else:
-                    env_spec_str = file.attrs.get("env_spec")
-                    assert isinstance(env_spec_str, str)
-                    env_spec = EnvSpec.from_json(env_spec_str)
-                    self._observation_space = gym.make(env_spec).observation_space
+                assert "observation_space" in file.attrs.keys(), "Minari hdf5 datasets must contain an observation_space attribute."
```

Comment:
This will be a breaking change tho, we should fix all past datasets (?)

Reply:
I'll need to check but I think that all of the latest versions of the datasets compatible with `minari>=0.4.0` have the observation and action space attributes. I'll double check before we merge this PR